runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -226,10 +226,25 @@ public void mergeSegments(Layout currentLayout) throws OutrankedException {
 
             sealEpoch(currentLayout);
 
-            LayoutBuilder layoutBuilder = new LayoutBuilder(currentLayout);
-            newLayout = layoutBuilder
-                    .mergePreviousSegment(1)
-                    .build();
+            Predicate<Layout> shouldMergeSegments = layout -> {
```

Comment:
This predicate is inconsistent with the if check in line 223. In that line If the segment size is greater than 1, but first two segments are different, then you still goes inside, seal epoch and update layout. I think you should use the same predicate for line 223.

Reply:
Good point